src/psyclone/kernel_tools.py

```diff
@@ -161,8 +159,15 @@ def run(args):
 
     try:
         if args.gen == "alg":
-            # Generate algorithm
-            code = alg_gen.generate(args.filename, api=api)
+            # Generate algorithm code.
+            if api == "dynamo0.3":
+                # pylint: disable=import-outside-toplevel
+                from psyclone.domain.lfric.algorithm.alg_gen import generate
```

Comment:
As mentioned above, I think it would be better if alg_gen had a class, and we would call the `generate` method of that class.

Reply:
Oh I see - I hadn't grasped that point earlier. I think you're probably right. Perhaps we could have alternative constructors or something - one for where we're dealing with an existing Algorithm layer and one where we're creating it from Kernel Metadata. That feels like a new issue - #1771.

Reply:
Well, that wasn't exactly what I meant, but your conclusion in the ticket (base on on the other) is what I actually think is easier to do if the alg_gen.py file would contain a class instead of just a selection of functions.